dxp/liferay_global_event_handler.md

```diff
@@ -0,0 +1,213 @@
+
+Listen to a custom event hosted by the global event handler one time.
+
+This is the equivalent to `after` except the listener is immediately detached when it is executed.
+
+##### Code Example:
+
+```js
+Liferay.onceAfter(
+    '<%= portletDisplay.getId() %>:messagePosted',
+    function (event) {
+        <%= randomNamespace %>onMessagePosted(
+            response,
+            refreshPage
+        );
+    }
+);
+
+```
+
+### `Liferay.after`:
+
+Description from [yui3/event-target.js at master · liferay/yui3 · GitHub](https://github.com/liferay/yui3/blob/master/src/event-custom/js/event-target.js#L848):
+
+Subscribe to a custom event hosted by the global event handler.
+
+The supplied callback will execute after any listeners add via the subscribe method, and after the default function, if configured for the event, has executed.
+
+##### Code Example:
+
+```js
+Liferay.after('commerce:productAddedToCart', function (event) {
+	Liferay.Portlet.refresh('#p_p_id<portlet:namespace />');
+});
+```
+
+### `Liferay.before`:
+
+Description from: [yui3/event-target.js at master · liferay/yui3 · GitHub](https://github.com/liferay/yui3/blob/master/src/event-custom/js/event-target.js#L886)
+
+Executes the global event callback before a DOM event, custom event or method. If the first argument is a function, it is assumed the target is a method.
+
+### `Liferay.detach`
+
+Description from: [yui3/event-target.js at master · liferay/yui3 · GitHub](https://github.com/liferay/yui3/blob/master/src/event-custom/js/event-target.js#L362)
+
+Detach one or more listeners the from a specified event
+
+##### Code Example
+
+```js
+var onDestroyPortlet = function () {
+	delegateHandler.removeListener();
+
+	Liferay.detach('destroyPortlet', onDestroyPortlet);
+};
+
+Liferay.on('destroyPortlet', onDestroyPortlet);
+```
+
+### `Liferay.detachAll`
+
+Description from: [yui3/event-target.js at master · liferay/yui3 · GitHub](https://github.com/liferay/yui3/blob/master/src/event-custom/js/event-target.js#L476)
+
+Removes all listeners from the specified event. If the event type is not specified, all listeners from all hosted custom events will be removed.
+
+### `Liferay.publish`
+
+Description from: [yui3/event-target.js at master · liferay/yui3 · GitHub](https://github.com/liferay/yui3/blob/master/src/event-custom/js/event-target.js#L559)
+
+Creates a new custom event of the specified type. If a custom event by that name already exists, it will not be re-created. In either case the custom event is returned.
+
+##### Code Example:
+
+```js
+Liferay.publish('allPortletsReady', {
+	fireOnce: true
+});
+
+// On another file located somewhere on DXP
+
+Liferay.on('allPortletsReady', function (event) {
+	document.getElementById('<%= HtmlUtil.escape(scroll) %>').scrollIntoView();
+});
+```
+
+### `Liferay.fire`
+
+Description from: [yui3/event-target.js at master · liferay/yui3 · GitHub](https://github.com/liferay/yui3/blob/master/src/event-custom/js/event-target.js#L723)
+
```

Comment:
I would change this `[yui3/event-target.js at master · liferay/yui3 · GitHub]` because I find it confusing for the reader: are we talking about yui3? liferay/yui3 or GitHub?

Reply:
I pointed to our yui3 fork because I was guessing we have some fixes and behavior changed as `liferay-ckeditor` but it's okay just point to the original repo :)

Reply:
Removed, as the original path was mentioned on the document heading. As discussed on https://github.com/liferay/liferay-frontend-guidelines/pull/198#discussion_r485456671